src/llm/llm_executor.hpp

```diff
@@ -73,10 +73,16 @@ class LLMExecutorWrapper {
     std::atomic<bool> finishExecutorThread = false;
 
     static void run(LLMExecutor* llmExecutor, std::atomic<bool>* receivedEndSignal) {
+        const uint64_t printMetricsEveryNumberOfSteps = 10;
+        std::atomic_uint64_t stepCounter = 0;
```

Comment:
why atomic. one thread can change it
why int64 - it will never exceed 10
why divide modulo 10 ? step==10 should be good enough